sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -791,6 +792,112 @@ def notify_media_setting(logical_port_name, transceiver_dict,
+                            break
+                    elif str(physical_port) == port:
+                        optics_si_dict = g_optics_si_dict[GLOBAL_MEDIA_SETTINGS_KEY][keys]
+                        break
+
+            elif RANGE_SEPARATOR in keys:
+                if check_port_in_range(keys, physical_port):
+                    optics_si_dict = g_optics_si_dict[GLOBAL_MEDIA_SETTINGS_KEY][keys]
+
+            key_dict = {}
+            if SPEED_KEY in optics_si_dict:
+                if key in optics_si_dict[SPEED_KEY]:
+                    key_dict = optics_si_dict[SPEED_KEY]
+                    return  key_dict[key]
+                elif vendor_name_str in optics_si_dict[SPEED_KEY]:
+                    key_dict = optics_si_dict[SPEED_KEY]
+                    return  key_dict[vendor_name_str]
+                elif DEFAULT_KEY in optics_si_dict[SPEED_KEY]:
+                    key_dict = optics_si_dict[SPEED_KEY]
+                    default_dict = key_dict[DEFAULT_KEY]
+
+    optics_si_dict = {}
+    
+    if PORT_MEDIA_SETTINGS_KEY in g_optics_si_dict:
+        for keys in g_optics_si_dict[PORT_MEDIA_SETTINGS_KEY]:
+            if int(keys) == physical_port:
+                optics_si_dict = g_optics_si_dict[PORT_MEDIA_SETTINGS_KEY][keys]
+                break
+
+        if len(optics_si_dict) == 0:
+            if len(default_dict) != 0:
+                return default_dict
+            else:
+                helper_logger.log_error("Error: No values for physical port '{}'".format(physical_port))
+            return {}
+
+        key_dict = {}
+        if SPEED_KEY in optics_si_dict:
+            if key in optics_si_dict[SPEED_KEY]:
+                key_dict = optics_si_dict[SPEED_KEY]
+                return  key_dict[key]
+            elif vendor_name_str in optics_si_dict[SPEED_KEY]:
+                key_dict = optics_si_dict[SPEED_KEY]
+                return  key_dict[vendor_name_str]
+            elif DEFAULT_KEY in optics_si_dict[SPEED_KEY]:
+                key_dict = optics_si_dict[SPEED_KEY]
+                default_dict = key_dict[DEFAULT_KEY]
+            elif len(default_dict) != 0:
+                return default_dict
+
+    return default_dict
+
+def fetch_optics_si_setting(logical_port_name, lane_speed, port_mapping):
+    if not g_optics_si_dict:
+        return
+
+    transceiver_dict = {}
+    optics_si = {}
+
+    physical_port_list = port_mapping.logical_port_name_to_physical_port_list(logical_port_name)
+    if physical_port_list is None:
+        helper_logger.log_error("Error: No physical ports found for logical port '{}'".format(logical_port_name))
+        return PHYSICAL_PORT_NOT_EXIST
+
+    for physical_port in physical_port_list:
+        if not _wrapper_get_presence(physical_port):
+            helper_logger.log_info("Module {} presence not detected during notify".format(physical_port))
+            continue
+        port_info_dict = _wrapper_get_transceiver_info(physical_port)
```

Comment:
This is an expensive operation since we will read multiple fields related to TRANCEIVER_INFO.
Since we use very few fields from the dict returned by _wrapper_get_transceiver_info, can we create a separate API to read fields required specifically for retrieving SI values?

Reply:
Again this was referred from  notify_media_setting(). I will create new api to retrieve only the values that I would need. Thanks

Reply:
Done